src/app/clusters/relative-humidity-measurement-server/relative-humidity-measurement-server.cpp

```diff
@@ -0,0 +1,110 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "relative-humidity-measurement-server.h"
+
+#include <app/util/af.h>
+
+#include "gen/att-storage.h"
+#include "gen/attribute-id.h"
+#include "gen/attribute-type.h"
+#include "gen/cluster-id.h"
+#include "gen/command-id.h"
+
+#include <support/logging/CHIPLogging.h>
+
+#ifndef emberAfRelativeHumidityMeasurementClusterPrintln
+#define emberAfRelativeHumidityMeasurementClusterPrintln(...) ChipLogProgress(Zcl, __VA_ARGS__);
+#endif
+
+EmberAfStatus emberAfRelativeHumidityMeasurementClusterGetMeasuredValue(chip::EndpointId endpoint, uint16_t * measuredValue)
+{
+    return emberAfReadServerAttribute(endpoint, ZCL_RELATIVE_HUMIDITY_MEASUREMENT_CLUSTER_ID,
+                                      ZCL_RELATIVE_HUMIDITY_MEASURED_VALUE_ATTRIBUTE_ID, (uint8_t *) measuredValue,
+                                      sizeof(*measuredValue));
+}
+
+EmberAfStatus emberAfRelativeHumidityMeasurementClusterGetMinMeasuredValue(chip::EndpointId endpoint, uint16_t * minMeasuredValue)
+{
+    return emberAfReadServerAttribute(endpoint, ZCL_RELATIVE_HUMIDITY_MEASUREMENT_CLUSTER_ID,
+                                      ZCL_RELATIVE_HUMIDITY_MIN_MEASURED_VALUE_ATTRIBUTE_ID, (uint8_t *) minMeasuredValue,
+                                      sizeof(*minMeasuredValue));
+}
+
+EmberAfStatus emberAfRelativeHumidityMeasurementClusterGetMaxMeasuredValue(chip::EndpointId endpoint, uint16_t * maxMeasuredValue)
+{
+    return emberAfReadServerAttribute(endpoint, ZCL_RELATIVE_HUMIDITY_MEASUREMENT_CLUSTER_ID,
+                                      ZCL_RELATIVE_HUMIDITY_MAX_MEASURED_VALUE_ATTRIBUTE_ID, (uint8_t *) maxMeasuredValue,
+                                      sizeof(*maxMeasuredValue));
+}
+
+void emberAfRelativeHumidityMeasurementClusterServerInitCallback(chip::EndpointId endpoint)
+{
+    uint16_t value       = 0;
+    EmberAfStatus status = emberAfRelativeHumidityMeasurementClusterGetMeasuredValue(endpoint, &value);
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        status = emberAfRelativeHumidityMeasurementClusterSetMeasuredValueCallback(endpoint, 0);
+        if (EMBER_ZCL_STATUS_SUCCESS != status)
+        {
+            emberAfRelativeHumidityMeasurementClusterPrintln("ERR: writing present value %x", status);
+        }
+    }
+
+    status = emberAfRelativeHumidityMeasurementClusterGetMinMeasuredValue(endpoint, &value);
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        status = emberAfRelativeHumidityMeasurementClusterSetMinMeasuredValueCallback(endpoint, UINT16_MAX);
```

Comment:
Please add a constant indicating that `UINT16_MAX` means "undefined". Also, couldn't those defaults be specified in the ZAP configuration file?

Reply:
I have removed the init impl. because the default values are stored via the attribute storage